### ISSUE TITLE
HTTP to HTTPS not working as it should

### ISSUE THREAD

**Issue by novastream** [aws-samples/aws-scalable-big-blue-button-example#62]
**Describe the bug**
At the moment we have a single BBB installation on a VPS that is working great but we needed to scale up and decided to run this script. Everything seems to be working as it should, it's way snappier than our VPS 👍 

I have read in the Scalelite repo that they fixed http to https redirects in may 2020. But it dosen't seem to be working in our case. When we try to browse http://conference.domain.com we get a a timeout error after 20-30 seconds.

**To Reproduce**
1. Go to http://conference.domain.com
2. Wait

**Expected behavior**
When going to http://conference.domain.com nginx should redirect to https://conference.domain.com.

**Desktop (please complete the following information):**
 - OS: [e.g. iOS]: Windows 10 20H2
 - Browser [e.g. chrome, safari]: Edge (Chromium), Brave and Firefox

**Comment by novastream** [comment]
I don't know if this is the right way but we sort of figured out a solution.

1. Go to EC2 and select Load Balancer
2. Create 2 new rules that takes port 80 traffic and redirects it to port 443 (one for each load balancer)
3. Go to Security Groups
4. Click on the Frontend ELB Security Group
5. Create a new inbound rule to allow traffic on port 80
6. Test, it should work immediately

**Comment by couchgott** [comment]
Hi @novastream, 
yes you're right. the current code does only listen on port 443. 
let me take this as a feature request. 

According the load balancers. you should only add the redirection rules and port 80 setup to the frontend load balancer. the 2nd load balancer is only called for the BBB API and is used by Greenlight to find the App Instance to redirect users too. so there won't be any users using that LB

**Comment by novastream** [comment]
Thank you for your input, we will adjust accordingly 👍

**Comment by ayuryogalifein** [comment]
Recordings in Greenlight points to http://scalelite.yourdomain.com. and here the conversion to https from http does not happen. How to solve this?

**Comment by novastream** [comment]
> 
> 
> Recordings in Greenlight points to http://scalelite.yourdomain.com. and here the conversion to https from http does not happen. How to solve this?

How to reproduce? Just so I can test as well :)

**Comment by novastream** [comment]
> 
> 
> > Recordings in Greenlight points to http://scalelite.yourdomain.com. and here the conversion to https from http does not happen. How to solve this?
> 
> How to reproduce? Just so I can test as well :)

Found it 👍 

**How to reproduce**
1. Start a chat room
2. Record something (audio/video or both)
3. Exit the chat room and wait for the recording to appear in the list of recordings
4. Click on the recording, it will open in a new tab (http) and give you a connection failed error
http://scalelite.domain.com/playback/presentation/2.0/playback.html?meetingId=xxxxxx 

5. The origin page will throw a mixed content error since the thumbnails load over http.
http://scalelite.domain.com/presentation/xxxx-xxxx/presentation/xxxx-xxxx/thumbnails/thumb-1.png

**Comment by SirMysterion** [comment]
I also found that the recordings link didn't have https. not sure how Greenlight is building the links.

Patching both load balancers to include a http to https redirect seems to work.

How is Greenlight building the URL? From the API reference?

**Comment by novastream** [comment]
> 
> 
> I also found that the recordings link didn't have https. not sure how Greenlight is building the links.
> 
> Patching both load balancers to include a http to https redirect seems to work.
> 
> How is Greenlight building the URL? From the API reference?

We just remove the one http to https forwarding rule per @couchgott suggestion. Will it work if we revert it back so both LBs have a forwarding rule?

**Comment by SirMysterion** [comment]
Just removed a rule? When was that?

**Comment by novastream** [comment]
> 
> 
> Hi @novastream,
> yes you're right. the current code does only listen on port 443.
> let me take this as a feature request.
> 
> According the load balancers. you should only add the redirection rules and port 80 setup to the frontend load balancer. the 2nd load balancer is only called for the BBB API and is used by Greenlight to find the App Instance to redirect users too. so there won't be any users using that LB

Just to avoid confussion, we did add two forwarding rules (one to each LB) and then as the commet quoted we removed one rule. 

But if I understand you correctly, by adding a forwarding rule to the second LB will fix the http error on recordings?

**Comment by SirMysterion** [comment]
> > Hi @novastream,
> > yes you're right. the current code does only listen on port 443.
> > let me take this as a feature request.
> > According the load balancers. you should only add the redirection rules and port 80 setup to the frontend load balancer. the 2nd load balancer is only called for the BBB API and is used by Greenlight to find the App Instance to redirect users too. so there won't be any users using that LB
> 
> Just to avoid confussion, we did add two forwarding rules (one to each LB) and then as the commet quoted we removed one rule.
> 
> But if I understand you correctly, by adding a forwarding rule to the second LB will fix the http error on recordings?

Yes, The BBBSc Load Balancer is for the API connection (https://scalelite.domain.com/bigbluebutton/api)
But the recording links also use scalelite.domain.com.

You will still have the mixed content http error. as the links generated by Greenlight are still http only.

**Comment by novastream** [comment]
> 
> 
> > > Hi @novastream,
> > > yes you're right. the current code does only listen on port 443.
> > > let me take this as a feature request.
> > > According the load balancers. you should only add the redirection rules and port 80 setup to the frontend load balancer. the 2nd load balancer is only called for the BBB API and is used by Greenlight to find the App Instance to redirect users too. so there won't be any users using that LB
> > 
> > 
> > Just to avoid confussion, we did add two forwarding rules (one to each LB) and then as the commet quoted we removed one rule.
> > But if I understand you correctly, by adding a forwarding rule to the second LB will fix the http error on recordings?
> 
> Yes, The BBBSc Load Balancer is for the API connection (https://scalelite.domain.com/bigbluebutton/api)
> But the recording links also use scalelite.domain.com.
> 
> You will still have the mixed content http error. as the links generated by Greenlight are still http only.

Thanks, I will do some testing tomorrow to see if we can get solid solution or a for the time being workaround :)

**Comment by ayuryogalifein** [comment]
I have tried adding the rule to the second LB. However, it does not work

**Comment by ayuryogalifein** [comment]
@novastream Did you get any solution for the presentation links?

**Comment by couchgott** [comment]
Hey folks, sorry for the delay working with the issue from my site. I am kind of knocked out (and in a hospital) right now. I will check and debug when back. It can be a change in the Big Blue Button code, default configs or greenlight/scalelite container images. Maybe a good idea to check for commits at these relates repos/registries and try a version before the breaking changes occur. 
I will try as soon as possible but currently i am not sure if it will be in a few days or a week. Sry for that

**Comment by ayuryogalifein** [comment]
Most probably the problem is with scalelite. https://github.com/blindsidenetworks/scalelite/pull/139.

In this case we may need to update /nginx/conf.d/scalelite-proxy.template. Will check and update

**Comment by ayuryogalifein** [comment]
It was simple issue. Not sure how I have missed. You need to add to redirect rule to both ALB and update the associated security groups with port 80 access. That's all. Thanks a lot for all the help. It would be good If we can add these configuration to the current stack. No more changes required
